### PR TITLE
[CALCITE-3382] add TUMBLE to parser

### PR DESCRIPTION
**Pull request by amaliujia**
add TUMBLE to parser and rename TUMBLE to "$TUMBLE".

Note that this PR also needs to add a non-reserved keywords "TUMBLE" to make parser recognize it. `reference.md` is also updated.

### CODE REVIEW

File: core/src/main/codegen/templates/Parser.jj [hunk 1]
```diff
@@ -7106,6 +7142,7 @@ SqlPostfixOperator PostfixRowOperator() :
 |   < XML: "XML" >
 |   < YEAR: "YEAR" >
 |   < ZONE: "ZONE" >
+|   <TUMBLE: "TUMBLE">
```
Comment (julianhyde) [inline core/src/main/codegen/templates/Parser.jj]: https://twitter.com/julianhyde/status/1174049701496143872 is all

Comment (amaliujia) [reply core/src/main/codegen/templates/Parser.jj]: oops. Updated.

**Comment by amaliujia** [comment]
@julianhyde 

Ok I have finished this PR and all tests have passed. Can you take a look?

File: core/src/main/java/org/apache/calcite/sql/fun/SqlStdOperatorTable.java [hunk 2]
```diff
@@ -2252,9 +2252,20 @@ public void unparse(
...
+   * "$" means TUMBLE will not be looked up by name. The real motivation of adding
+   * TUMBLE to parser is that, we are planning to support TUMBLE as a table
+   * function(see [CALCITE-3272]). It turns out that it will require non-trivial changes
+   * to reuse "TUMBLE" as operator names for both TUMBLE group function and TUMBLE
+   * table function. Thus the resolution is to let parser recognize TUMBLE group function
+   * so that TUMBLE group function can be renamed. "TUMBLE" as a name will only be used by
+   * TUMBLE table function. This design also makes TUMBLE group function deprecation easier
+   * as TUMBLE group function should be deprecated after its replacement(TUMBLE table
+   * function) is mature. See [CALCITE-3340] for more context about this design.
+   */
```
Comment (amaliujia) [inline core/src/main/java/org/apache/calcite/sql/fun/SqlStdOperatorTable.java]: @julianhyde  added more context here.

**Comment by amaliujia** [comment]
A friendly ping to raise attention on this PR.

**Pull request closed without merging**